modules/dhcp/subnet_service.rb

```diff
@@ -93,13 +97,13 @@ def find_host_by_ip(subnet_address, ip_address)
     @reservations_by_ip[subnet_address, ip_address]
   end
 
-  def find_host_by_hostname(subnet_address, hostname)
-    @reservations_by_name[subnet_address, hostname]
+  def find_host_by_hostname(hostname, subnet_address = nil)
```

Comment:
The only place where find_host_by_hostname was used was https://github.com/theforeman/smart-proxy/blob/develop/modules/dhcp/providers/server/isc.rb#L114, which you changed. You can drop subnet_address from the method parameters now; all_hosts method can use either reservations_by_ip or reservations_by_mac to return host records for a subnet. After that you can get rid of reservations_by_subnet :).